doc/build_ref/TrilinosBuildReferenceTemplate.rst

```diff
@@ -105,3 +105,65 @@ In order do co-development of TriBTS and Trilinos (see http://http://trac.trilin
 
 (NOTE: You have to use the data-type ``STRING`` with ``Trilinos_TRIBITS_DIR``
 or CMake will automatically assume it is relative to the build dir!)
+
+
+Example: Configuring with Kokkos and advanced back-ends
+--------------------------------------------------------
+
+Kokkos serves as a basis for many advanced trilinos packages, especially those
+wanting to take advantages of advanced hardware such as the latest GPU.  
+Kokkos has its own build system that can be invoked, and this build system has
+more options than that supported by Trilinos by default, as it includes a rich
+set of options.   If Trilinos has the ability to set a Kokkos setting, that will
+be used, otherwise one has to use the Kokkos setting.  Below we list which
+options to use::
+
+  +----------------------------+----------------------------------------------+
+  | Functionality              | Option                                       |
+  +============================+==============================================+
+  | Specify architecture       | KOKKOS_HOST_ARCH                             |
+  +----------------------------+----------------------------------------------+
+  | Build with separate libs   | KOKKOS_SEPARATE_LIBS                         |
+  +----------------------------+----------------------------------------------+
+  | Debug builds               | KOKKOS_DEBUG                                 |
+  +----------------------------+----------------------------------------------+
+  | Devices Options                                                           |
```

Comment:
Grammar fix: "Device options" (note capitalization change, for consistency with the other column headers).

Reply:
Done